### ISSUE TITLE
10.2 RC1: Possible usability issue with OAuth: cant't add app password

### ISSUE THREAD

**Issue by thommierother** [owncloud/core#35149]
Maybe I am using the wrong steps to reproduce this but here is my observation:

environment:
Server: 10.2 RC1 owncloud server, oauth app enabled
Sync client: sync client 2.5.4 build 515 (Linux)

I want to add a sync account with a permanent app password INSTEAD of going through the oauth browser auth steps upon every start of the sync application.

1. create app password in the oc account
2. add new account in the client
3. enter server address
4. client redirects to oauth authorization screen
5. oauth login successful, BUT:

My problem is that there is no option to add the app password INSTEAD on the client side. Not in the client application itself and not on the oath consent screen presented by the server. Or am I totally wrong?

two possible options:
a) entry  in the sync client to add app password application AFTER doing the Oath process one time
b) additional button in the Oauth consent screen to do this DURING the first Oauth process

**Comment by PVince81** [comment]
@thommierother not sure if oauth and app password are compatible.

usually the app password can be added directly in place of the user password when entering credentials. So instead of "user id" and "user password" just enter "user id" and "app password". Same on mobile.

@DeepDiver1975

**Comment by thommierother** [comment]
Yes I know, but I think of a use case where a user has oauth active for web access and uses the sync client in parallel. In the moment active oauth blocks the usage of automatic sync. I see no reason why it should not be possible to combine both auth methods on the same user account. In the log run, users could also use other identity providers for oauth access through the web interface...

**Comment by patrickjahns** [comment]
@thommierother 
Please correct me if I am mistaken - but the behavior you are describing is not solely tied to 10.2RC1 - it should be the same on previous ownCloud server versions with oauth2 app installed.

Let me quickly rephrase the issue - and what I understood you want to achieve - please correct this if I missed the point.

At the baseline, you are having ownCloud with oauth app installed. Now you want to connect a specific client via the app password/token feature. 
However, when you let the client connect, it will trigger the oauth flow ( Authorize the client on the ownCloud server side). 
In that scenario, you are no longer able to enter "somewhere" the username+password for the client.

This rather sounds to me like a usability issue of the client, as the client is deciding wether or not to use oauth or allow username+password. There is nothing the ownCloud server could do.
If that's the case - let's move this ticket over to https://github.com/owncloud/client/issues

**Comment by patrickjahns** [comment]
Note: I've removed it from the QA milestone - as I don't consider this to be blocking the upcoming release

**Comment by thommierother** [comment]
@patrickjahns Yes, you are correct. It is mainly an issue of the client.
